test/server/cards/02_SHD/upgrades/BrutalTraditions.spec.ts

```diff
@@ -0,0 +1,68 @@
+describe('Brutal Traditions', function() {
+    integration(function(contextRef) {
+        describe('Brutal Tradition\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: [
+                            { card: 'vigilant-honor-guards', upgrades: ['brutal-traditions'] },
+                            { card: 'moisture-farmer' },
+                            { card: 'death-trooper' }
+                        ],
+                    },
+                    player2: {
+                        groundArena: ['wampa'],
+                        hand: ['confiscate', 'vanquish', 'rivals-fall']
+                    }
+                });
+            });
+
+            it('should be able to play it from the discard pile when an opponent\'s unit is defeated.', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.vigilantHonorGuards);
+                context.player1.clickCard(context.wampa);
+
+                expect(context.player2).toBeActivePlayer();
+                expect(context.wampa).toBeInLocation('discard');
+
+                context.player2.clickCard(context.confiscate);
+                expect(context.brutalTraditions).toBeInLocation('discard');
+                expect(context.player1).toBeActivePlayer();
+                expect(context.player1).toBeAbleToSelect(context.brutalTraditions);
+
+                context.player1.clickCard(context.brutalTraditions);
+                context.player1.clickCard(context.vigilantHonorGuards);
+                expect(context.vigilantHonorGuards.upgrades).toEqual([context.brutalTraditions]);
+                expect(context.vigilantHonorGuards.getPower()).toBe(5);
+                expect(context.vigilantHonorGuards.getHp()).toBe(8);
+                expect(context.vigilantHonorGuards.damage).toBe(4);
```

Comment:
Can remove these checks, we have other tests confirming that attached upgrades give stats correctly. Just checking the upgrade is attached is enough